timemachine/cpp/CMakeLists.txt

```diff
@@ -32,6 +34,20 @@ endif()
 add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/${PYBIND_SRC_DIR} ${CMAKE_CURRENT_BINARY_DIR}/${PYBIND_SRC_DIR})
 
 
+# Disable warnings on vendored files
+file(
+  GLOB
+  VENDORED_FILES
+  "src/vendored/*.c*"
+  "src/vendored/*.h*"
+)
+set_source_files_properties(
+  ${VENDORED_FILES}
+  PROPERTIES
+  COMPILE_FLAGS "-w"
```

Comment:
Note: the [CMake docs for `COMPILE_FLAGS`](https://cmake.org/cmake/help/latest/prop_sf/COMPILE_FLAGS.html) mention

> Note This property has been superseded by the [COMPILE_OPTIONS](https://cmake.org/cmake/help/latest/prop_sf/COMPILE_OPTIONS.html#prop_sf:COMPILE_OPTIONS) property.

Reply:
Assuming this applies to both C++ and CUDA?

Reply:
Having added a small test file in the vendored with the extension `.cu`, it appears to correctly suppress warnings.